server/fleet/microsoft_mdm.go

```diff
@@ -820,77 +823,373 @@ func (e MDMWindowsEnrolledDevice) AuthzType() string {
 }
 
 ///////////////////////////////////////////////////////////////
-/// Microsoft MS-MDM message
+// Microsoft MS-MDM message
+// MS-MDM is a client-to-server protocol that consists of a SOAP-based Web service.
+// MDM is based on the OMA-DM protocol. Messages are issued by a requester and results and status are returned by a responder as a SynCML message.
+// A SyncML message is a well-formed XML document that adheres to the document type definition (DTD), but which does not require validation.
+// The XML document is identified by a SyncML document element type that serves as a parent container for the SyncML message.
+// The SyncML message consists of a header specified by the SyncHdr  element type and a body specified by the SyncBody element type.
+// The SyncML header identifies the routing and versioning information about the SyncML message.
+// The SyncML body functions as a container for one or more SyncML commands.
+// A SyncML command is specified by individual element types that provide specific details about the command, including any data or meta-information.
+// MS-MDM uses a subset of the SyncML message definition specified in OMA-SyncMLRP spec. MDM-specific SyncML xml message format is defined in OMA-DMRP.
 
-type SyncMLMessage struct {
-	XMLinfo string       `xml:"xmlns,attr"`
-	Header  SyncMLHeader `xml:"SyncHdr"`
-	Body    SyncMLBody   `xml:"SyncBody"`
+type SyncML struct {
+	XMLName  xml.Name `xml:"SyncML"`
+	Xmlns    string   `xml:"xmlns,attr"`
+	SyncHdr  SyncHdr  `xml:"SyncHdr"`
+	SyncBody SyncBody `xml:"SyncBody"`
 }
 
-// SyncML XML Parsing Types - This needs to be improved
-type SyncMLHeader struct {
-	DTD        string `xml:"VerDTD"`
-	Version    string `xml:"VerProto"`
-	SessionID  int    `xml:"SessionID"`
-	MsgID      int    `xml:"MsgID"`
-	Target     string `xml:"Target>LocURI"`
-	Source     string `xml:"Source>LocURI"`
-	MaxMsgSize int    `xml:"Meta>A:MaxMsgSize"`
+type SyncHdr struct {
+	VerDTD    string   `xml:"VerDTD"`
+	VerProto  string   `xml:"VerProto"`
+	SessionID string   `xml:"SessionID"`
+	MsgID     string   `xml:"MsgID"`
+	Target    *LocURI  `xml:"Target,omitempty"`
+	Source    *LocURI  `xml:"Source,omitempty"`
+	Meta      *MetaHdr `xml:"Meta,omitempty"`
 }
 
-type SyncMLCommandMeta struct {
-	XMLinfo string `xml:"xmlns,attr"`
-	Type    string `xml:"Type"`
+type MetaHdr struct {
+	MaxMsgSize *string `xml:"MaxMsgSize,omitempty"`
 }
 
-type SyncMLCommandItem struct {
-	Meta   SyncMLCommandMeta `xml:"Meta"`
-	Source string            `xml:"Source>LocURI"`
-	Data   string            `xml:"Data"`
+// ProtoCmds contains a slice of SyncML protocol commands
+type ProtoCmds *[]SyncMLCmd
```

Comment:
This probably shouldn't be a pointer  to a slice, as a slice can be `nil` so it can represent absence.

Reply:
Thanks, I just updated this one. The main idea was to represent absence during JSON conversion to go type.